docker-build-proto/Dockerfile

```diff
@@ -0,0 +1,46 @@
+FROM golang:1.14-alpine3.11 as builder
+
+RUN apk add --update --no-cache build-base curl git upx && \
+  rm -rf /var/cache/apk/*
+
+ENV GOLANG_PROTOBUF_VERSION=1.3.1 \
+  GOGO_PROTOBUF_VERSION=1.3.1
+RUN GO111MODULE=on go get \
+  github.com/golang/protobuf/protoc-gen-go@v${GOLANG_PROTOBUF_VERSION} \
+  github.com/gogo/protobuf/protoc-gen-gogo@v${GOGO_PROTOBUF_VERSION} && \
+  mv /go/bin/protoc-gen-go* /usr/local/bin/
+
+ENV PROTOBUF_VERSION=3.11.4
+RUN mkdir -p /tmp/protoc && \
+  curl -sSL \
+  https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOBUF_VERSION}/protoc-${PROTOBUF_VERSION}-linux-x86_64.zip \
+  -o /tmp/protoc/protoc.zip && \
+  cd /tmp/protoc && \
+  unzip protoc.zip && \
+  mv /tmp/protoc/include /usr/local/include
+
+RUN upx --lzma /usr/local/bin/*
+
+FROM alpine:edge
```

Comment:
did not know multiple FROM are allowed, cool!

Reply:
super useful to keep images small: https://matthiasnoback.nl/2017/04/docker-build-patterns/